aperturedb/ParallelQuery.py

```diff
@@ -257,6 +260,13 @@ def process_responses(requests, input_blobs, responses, output_blobs):
                 response_handler = self.generator.response_handler
             if hasattr(self.generator, "strict_response_validation") and isinstance(self.generator.strict_response_validation, bool):
                 strict_response_validation = self.generator.strict_response_validation
+
+            # if response_handler doesn't support index, just discard the index with a wrapper.
+            if response_handler is not None:
+                if not 'index' in inspect.signature(response_handler).parameters:
```

Comment:
it wouldn't work. also if they happened to have a 4 query version and called the first parameter 'index' it would not wrap it.

Perhaps we just want to see if parameters is greater than 4?

Reply:
> Perhaps we just want to see if parameters is greater than 4?

That might be better.

Reply:
done